### DISCUSSION TITLE
Hugo book theme and markdown creation for Hugo

### DISCUSSION THREAD

**Discussion by mrchristian** [singlesourcepub/community#8] [Q&A]
Couple of linked questions:

1. Anyone used the Hugo book theme https://themes.gohugo.io/hugo-book/ ? I'm asking as thinking of trying out for some student books.
2. If I wanted to convert a director of HTML to MD for use with the theme, so I got page header to set doc order in MD, etc., what would people recommend as a workflow to batch convert the HTML, with pics. Here is the sample HTML https://github.com/TIBHannover/open-science-guide-of-guides/tree/main/html

thanks

Its for a student project and I'd like to have them put a website on book front end ;-)

**Comment by xldrkp** [comment]
Hi! I never used it, but I'm very happy with https://github.com/matcornic/hugo-theme-learn/ for my student books. See https://www3.tuhh.de/itbh/itbh-informatik-2020/veranstaltungsskript-informatik-2020_21/ for an example.

For me, there are two alternatives to use for book-like structures: First, https://www.docsy.dev/ runs on Hugo and has all the bells and whistles that comes with it. But for me some things are a bit overloaded so that I tried https://squidfunk.github.io/mkdocs-material/ which runs on Python, has a similar page architecture but is much easier to handle. Careful, it's an "insider modell". Means, there are some features only available for paying users. That's fine for me, no prob with that. It has all features I need in the free version.

To convert your HTML for your book, I suggest to use https://pandoc.org/.

**Reply by jcolomb** [reply]
Bookdown is a nice way to create books without external service (runs with R). There is a python equivalent used for example by the turing way book.
But you know that right?

**Reply by mrchristian** [reply]
Hi Julian,

Bookdown, I remember you showing me around it, although I'm not a daily user. In the SSP community there will be people from very different communities and technical stacks, so most likely people who are not familiar with Bookdown.

Idea! He starts list of platforms workflows.

Maybe you'd like to join us one time for Gin-Tonic https://genr.eu/wp/towards-a-standardized-research-folder-structure/ and show people around Bookdown and Gin-Tonic :-)

**Comment by mrchristian** [comment]
Learn theme I like it, it has magic https://learn.netlify.app/en/basics/requirements/ ;-)

Do your students do any styling?

And for Padoc HTML conversion - is there a good place to start with a tutorial for this? When I just 'go at it' and try to make it work I fail :-(

Onwards and up

**Reply by xldrkp** [reply]
@mrchristian My students do not work with the theme, I prepare the course content with it. I once tried to motivate them in participating but the learning curve for Git(Lab|Hub) workflows is too steep in that semester.

**Reply by xldrkp** [reply]
@mrchristian I gave the `html/document-1.html` a try with `pandoc document-1.html -o document-1.md` and got a nice Markdown document. You will have to add some YAML frontmatter stuff but I guess you can go with that.
